src/domain/certificates/useCases/downloadPDF.ts

```diff
@@ -0,0 +1,176 @@
+    height: PDF_COVER_PAGE_HEIGHT,
+    margin: 0
+  });
+
+  deleteDomContainer(htmlContainer);
+
+  return pdf;
+};
+
+const extractHtmlLayoutFromContainer = (htmlLayout: HTMLElement, isLayoutLegacy: boolean): HTMLElement[] => {
+  // DOM Path is static, always the same (set in legacy display)
+  return isLayoutLegacy
+    ? [htmlLayout]
+    : Array.from(htmlLayout.querySelectorAll(':scope > div > div'));
+};
+
+const generateLayoutPages = async (pdf: jsPDF, pagesContent: string): Promise<jsPDF> => {
+  const isLayoutLegacy: boolean = displayHtmlHasNoWidthConstraint(pagesContent);
+  const htmlContainer: HTMLElement = createDomContainer({
+    ...HTML2PDF_CONTAINER_STYLE,
+    ...HTML2PDF_CONTAINER_HIDE_STYLE,
+    ...(isLayoutLegacy ? HTML2PDF_CONTAINER_LEGACY_STYLE : {})
+  });
+
+  htmlContainer.insertAdjacentHTML('beforeend', pagesContent);
+
+  const pages: HTMLElement[] = extractHtmlLayoutFromContainer(htmlContainer, isLayoutLegacy);
+
+  for (const page of pages) {
+    const pageSize = getHTMLElementSize(page);
+    pdf = await addHTMLPageToPDF({
+      pdf,
+      pageHTMLElement: page,
+      width: pageSize.width,
+      height: pageSize.height,
+      margin: PDF_MARGIN
+    });
+  }
+
+  deleteDomContainer(htmlContainer);
+
+  return pdf;
+};
+
+const createDomContainer = (containerStyles?: TCSSPropertiesObject): HTMLElement => {
+  const html2pdfDomContainer: HTMLElement = document.createElement('div');
+  html2pdfDomContainer.setAttribute('id', HTML2PDF_DOM_CONTAINER_ID);
+
+  if (containerStyles) {
+    addCSSPropertiesToHTMLElement(html2pdfDomContainer, containerStyles);
+  }
+
+  document.body.appendChild(html2pdfDomContainer);
+
+  return html2pdfDomContainer;
+};
+
+const deleteDomContainer = (container: HTMLElement): void => {
+  document.body.removeChild(container);
+};
+
+const generatePdfDocument = async ({
+  contentType = null,
+  contentEncoding = '',
+  content = '',
+  recipientName = '',
+  certificateTitle = '',
+  issueDate = '',
+  issuerName = '',
+  issuerLogo = '',
+  recordLink = '',
+  issuerPublicKey = ''
+}: IDownloadPdfParameters): Promise<void> => {
+  const fileName: string = certificateTitle;
+  let pdf: jsPDF;
+
+  const isContentAnImage: boolean = isContentTypeImage(contentType);
+  const isContentHTML: boolean = contentType === CONTENT_TYPES.TEXT_HTML;
+  const shouldHaveCoverPage: boolean = isContentHTML || isContentAnImage;
+
+  if (shouldHaveCoverPage) {
+    pdf = await generateCoverPage(recipientName, certificateTitle, issueDate, issuerName, issuerLogo, recordLink, issuerPublicKey);
+  }
+
+  if (isContentHTML) {
+    pdf = await generateLayoutPages(pdf, content);
+  } else if (isContentAnImage) {
+    pdf = await generatePdfFromImage(pdf, `data:${contentType};${contentEncoding},${content}`);
+  } else {
+    const extension: string = getFileExtensionFromContentType(CONTENT_TYPES.APPLICATION_PDF);
+    forceDownloadFile(content, contentType, contentEncoding, fileName, extension);
```

Comment:
wait, if it's a pdf we should have pdf viewer on the page already, does that not offer native download?

Reply:
It does have it, but I figured this is consistent and will probably be more visible for all users, especially the less technical savvy ones.